src/models/apps/expense-split-app/expensegroup.model.js

```diff
@@ -0,0 +1,55 @@
+import mongoose, { Schema } from "mongoose";
+import {
+  AvailableExpenseGroupTypes,
+  ExpenseGroupTypes,
+} from "../../../constants.js";
+
+const expenseGroupSchema = new mongoose.Schema(
+  {
+    name: {
+      type: String,
+      required: true,
+    },
+    description: {
+      type: String,
+    },
+    groupOwner: {
+      type: Schema.Types.ObjectId,
+      ref: "User",
+    },
+
+    participants: [
+      {
+        type: Schema.Types.ObjectId,
+        ref: "User",
+      },
+    ],
+    groupCategory: {
+      type: String,
+      enum: AvailableExpenseGroupTypes.OTHERS,
+      default: ExpenseGroupTypes.OTHERS,
+      required: true,
+    },
+    groupTotal: { type: Number, default: 0 },
+    split: {
+      type: Map,
```

Comment:
why did you use `Map` here? Please explain

Reply:
no specific reason trying to optimise the data structure and its functions of get delete or set is self explanatory and was learning about map at that time so tried to optimise a project using map in place of key value pairs